src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -493,49 +463,126 @@ def handleSuccessful(self, jobID, fwkJobReport, fwkJobReportPath = None):
                                                 conn = self.getDBConn(),
                                                 transaction = self.existingTransaction())
 
-        fileList = fwkJobReport.getAllFiles()
-        
-        bookKeepingSuccess = True
-        
-        for fwjrFile in fileList:
-            # associate logArchived file for parent jobs on wmstats assuming fileList is length is 1.
-            if jobType == "LogCollect":
+        if jobSuccess:
+            fileList = fwkJobReport.getAllFiles()
+
+            # consistency check comparing outputMap to fileList
+            # they should match except for some limited special cases
+            outputModules = set([])
+            for fwjrFile in fileList:
+                outputModules.add(fwjrFile['outputModule'])
+            if set(outputMap.keys()) == outputModules:
+                pass
+            elif jobType == "LogCollect" and len(outputMap.keys()) == 0 and outputModules == set(['LogCollect']):
+                pass
+            elif jobType == "Merge" and set(outputMap.keys()) == set(['Merged', 'MergedError', 'logArchive']) and outputModules == set(['Merged', 'logArchive']):
+                pass
+            elif jobType == "Merge" and set(outputMap.keys()) == set(['Merged', 'MergedError', 'logArchive']) and outputModules == set(['MergedError', 'logArchive']):
+                pass
+            elif jobType == "Express" and set(outputMap.keys()).difference(outputModules) == set(['write_RAW']):
+                pass
+            else:
+                failJob = True
+                if jobType in [ "Processing", "Production" ]:
+                    cmsRunSteps = 0
+                    for step in fwkJobReport.listSteps():
+                        if step.startswith("cmsRun"):
+                            cmsRunSteps += 1
+                    if cmsRunSteps > 1:
+                        failJob = False
+
+                if failJob:
+                    jobSuccess = False
+                    logging.error("Job %d , list of expected outputModules does not match job report, failing job", jobID)
+                    logging.debug("Job %d , expected outputModules %s", jobID, sorted(outputMap.keys()))
+                    logging.debug("Job %d , fwjr outputModules %s", jobID, sorted(outputModules))
+                    fileList = fwkJobReport.getAllFilesFromStep(step = 'logArch1')
+                else:
+                    logging.debug("Job %d , list of expected outputModules does not match job report, accepted for multi-step CMSSW job", jobID)
+        else:
+            fileList = fwkJobReport.getAllFilesFromStep(step = 'logArch1')
+
+        if jobSuccess:
+            logging.info("Job %d , handle successful job", jobID)
+        else:
+            logging.error("Job %d , bad jobReport, failing job",  jobID)
```

Comment:
Is this really an error (bad job report)? It could be that the job didn't have the same output modules and in this case it's a failed job but not bad job report, right?
In other words, are we going to print an ERROR in logs for any/every job that has exit code != 0? If the answer is yes, the we should change this logging.